README.md

```diff
@@ -10,29 +10,49 @@ This repo contains the AWS ParallelCluster Chef cookbook used in AWS ParallelClu
-- `aws-parallelcluster-environment` contains the resources and recipes related to AWS services and their configuration
-- `aws-parallelcluster-computefleet` contains the resources and recipes related to scaling logic
-- `aws-parallelcluster-awsbatch` contains the resources and recipes required to support AWS Batch as scheduler
-- `aws-parallelcluster-slurm` contains the resources and recipes required to support Slurm as scheduler
-- `aws-parallelcluster-tests` contains the resources and recipes tools used for Inspec tests
-
-The `cookbooks/third-party` folder contains cookbooks from marketplace. 
-You can find more information about them in the `cookbooks/third-party/THIRD-PARTY-LICENSES.txt` file.
+the other sub-cookbooks and third party cookbooks.
+
+The main cookbook does not contain any recipe, attribute or library. They are distributed in the functional cookbooks under the `cookbooks` folder
+defined as follows:
+- `aws-parallelcluster-entrypoints` is the cookbook to define the external interface, contains recipes called by AMI builder, cluster setup, cluster update, etc. 
+  The recipes in this cookbook are called directly by the CLI or CI/CD. It orchestrates invocation of recipes/resources from other cookbooks;
+- `aws-parallelcluster-platform` OS packages and system configuration (directories, users, services, drivers);
+- `aws-parallelcluster-environment` AWS services configuration and usage, such as shared file systems, directory service and network interfaces;
+- `aws-parallelcluster-computefleet` scaling logic, compute fleet scripts and daemons;
+- `aws-parallelcluster-awsbatch` files required to support AWS Batch as scheduler;
+- `aws-parallelcluster-slurm` files required to support Slurm as a scheduler and its dependencies (Munge, MySQL for accounting, etc);
+
+Finally, some common code, such as source/script directories, usernames, package installer etc., are located in a cookbook 
+which every other cookbook depend on, that is `aws-parallelcluster-shared`.
+Each cookbook hosts recipes and resources, attributes, functions, files and templates belonging to its functional area.
+
+Every cookbook contains ChefSpec and Kitchen tests for its code.
+However, the code in a cookbook might require that some other code from a different cookbook not listed among dependencies,
+to be executed as a prerequisite (test setup phase). For this reason `aws-parallelcluster-tests` cookbook must depend on every other cookbook.
+
+The `test` folder contains Python unit tests files and Kitchen [environment](https://docs.chef.io/environments/) files.
 
 The `kitchen` folder contains utility files used when running Inspec tests locally.
 
-The `test` folder contains Python unit tests files and Kitchen [environment](https://docs.chef.io/environments/) files.
+The `cookbooks/third-party` folder contains cookbooks from marketplace. They must be regularly updated and should not be modified by hand.
```

Comment:
Should we specify why they have been pre-downloaded and stored in our repository instead of being vendored from marketplace?

Reply:
done.,